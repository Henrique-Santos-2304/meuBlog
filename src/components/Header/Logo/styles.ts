import styled, { css } from "styled-components";
export const Wrapper = styled.div`
  padding: 0.3rem 2rem;
  background: transparent;
`;

export const PreText = styled.p`
  ${({ theme }) => css`
    font-family: ${theme.font.family.Tangerine};
    font-size: 4.5rem;
    color: ${theme.colors.white};
    span {
      color: ${theme.colors.red800};
    }
  `}
`;
