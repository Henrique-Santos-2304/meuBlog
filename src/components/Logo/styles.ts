import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: max-content;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.xsmall};
    background: ${theme.colors.mainBg};
    border-radius: ${theme.border.radius};
  `}
`;
export const PreTextLogo = styled.p`
  ${({ theme }) => css`
    font-family: ${theme.font.family.Tangerine};
    font-size: 6rem;
    color: ${theme.colors.white};
  `}
`;

export const ProTextLogo = styled.p`
  ${({ theme }) => css`
    font-family: ${theme.font.family.Tangerine};
    font-size: 6rem;
    color: ${theme.colors.red800};
  `}
`;
