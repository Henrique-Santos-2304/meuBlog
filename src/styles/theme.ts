export default {
  grid: {
    container: "130rem",
    containerContent: "60rem",
    gutter: "3.2rem",
  },
  border: {
    radius: "1.2rem",
  },
  colors: {
    mainBg: " #12131C",
    gray: "#524861",
    white: "#E5E7EB",
    black700: "#111",
    black800: "#06092b",
    black900: "#000",
    red800: "#C11010",
  },
  font: {
    family: {
      Poppins: "'Poppins', sans-serif",
      Tangerine: "'Tangerine', cursive",
      GrechenFuemen: '"Grechen Fuemen", cursive',
      Roboto: '"Roboto", monospace',
    },
    sizes: {
      xsmall: "1.2rem",
      small: "1.4rem",
      medium: "1.6rem",
      large: "1.8rem",
      xlarge: "2.0rem",
      xxlarge: "2.8rem",
      xbiglarge: "3rem",
      xxbiglarge: "3.4rem",
    },
    light: 300,
    normal: 500,
    bold: 700,
  },
  spacings: {
    xxsmall: "0.8rem",
    xsmall: "1.6rem",
    small: "2.4rem",
    medium: "3.2rem",
    large: "4.0rem",
    xlarge: "4.8rem",
    xxlarge: "5.6rem",
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },
};
