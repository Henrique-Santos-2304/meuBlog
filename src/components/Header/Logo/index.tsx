import * as S from "./styles";
import { Box } from "@chakra-ui/react";

const Logo = () => (
  <S.Wrapper>
    <Box padding="4" maxW="3xl">
      <S.PreText aria-label="logo dev">
        Web<span>Dev</span>
      </S.PreText>
    </Box>
  </S.Wrapper>
);

export default Logo;
