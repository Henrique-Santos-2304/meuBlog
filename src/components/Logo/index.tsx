import * as S from "./styles";

const Logo = () => (
  <S.Wrapper>
    <S.PreTextLogo>Web</S.PreTextLogo>
    <S.ProTextLogo>Dev</S.ProTextLogo>
  </S.Wrapper>
);

export default Logo;
