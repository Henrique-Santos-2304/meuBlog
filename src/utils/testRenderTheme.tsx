import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
