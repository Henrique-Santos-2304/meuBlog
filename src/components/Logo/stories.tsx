import { Story, Meta } from "@storybook/react";
import Logo from ".";

export default {
  title: "Logo",
  component: Logo,
} as Meta;

export const Basic: Story = (args) => <Logo {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <Logo {...args} />;
