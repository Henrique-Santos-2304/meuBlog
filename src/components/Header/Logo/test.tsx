import Logo from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<Logo />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Logo />);
    expect(container).toMatchSnapshot();
  });
  it("should logo text in the display ", () => {
    renderTheme(<Logo />);
    expect(screen.getByLabelText(/logo dev/i)).toBeInTheDocument();
  });
  it("should have a text Web Dev  ", () => {
    renderTheme(<Logo />);
    expect(screen.getByText(/web/i)).toBeInTheDocument();
    expect(screen.getByText(/dev/i)).toBeInTheDocument();
  });
});
