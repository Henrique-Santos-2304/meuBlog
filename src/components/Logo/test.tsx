import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import theme from "styles/theme";
import "jest-styled-components";

import Logo from ".";

describe("<Logo />", () => {
  it("create Snapshot for element", () => {
    const { container } = renderTheme(<Logo />);
    expect(container).toMatchSnapshot();
  });
  it("should have a text Web", () => {
    renderTheme(<Logo />);
    expect(screen.getByText(/Web/i)).toBeInTheDocument();
    expect(screen.getByText(/Web/i)).toHaveStyle({ color: theme.colors.white });
  });
  it("should have a text Dev", () => {
    renderTheme(<Logo />);
    expect(screen.getByText(/Dev/i)).toBeInTheDocument();
    expect(screen.getByText(/Dev/i)).toHaveStyle({
      color: theme.colors.red800,
    });
  });
});
