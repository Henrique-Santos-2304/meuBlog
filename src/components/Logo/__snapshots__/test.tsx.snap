// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> create Snapshot for element 1`] = `
.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.6rem;
  background: #12131C;
  border-radius: 1.2rem;
}

.c1 {
  font-family: 'Tangerine',cursive;
  font-size: 6rem;
  color: #E5E7EB;
}

.c2 {
  font-family: 'Tangerine',cursive;
  font-size: 6rem;
  color: #C11010;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      Web
    </p>
    <p
      class="c2"
    >
      Dev
    </p>
  </div>
</div>
`;
